src/parse/converters/expressions/primary/literal/readRegexpLiteral.js

```diff
@@ -0,0 +1,17 @@
+import { REGEXP_LITERAL } from 'config/types';
+
+// as far as I can tell, this only falls over if modifiers are repeated erroneously
+var regexpPattern = /^\/(?:(?=\\\/)..|[^\/])+\/[gimy]{0,4}/;
```

Comment:
OK, it seems I was wrong. The regexp literal syntax isn't so complicated after all. This should cut it:
`/^\/(?:[^\n\r\u2028\u2029/\\[]|\\.|\[(?:[^\n\r\u2028\u2029\]\\]|\\.)*])+\/(?:([gimuy])(?![a-z]*\1))*(?![a-zA-Z_$0-9])/`